service/kg-search/src/main/java/eu/ebrains/kg/search/api/Preflight.java

```diff
@@ -0,0 +1,83 @@
+/*
+ *  Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
+ *  Copyright 2021 - 2023 EBRAINS AISBL
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  This open source software code was developed in part or in whole in the
+ *  Human Brain Project, funded from the European Union's Horizon 2020
+ *  Framework Programme for Research and Innovation under
+ *  Specific Grant Agreements No. 720270, No. 785907, and No. 945539
+ *  (Human Brain Project SGA1, SGA2 and SGA3).
+ */
+
+package eu.ebrains.kg.search.api;
+
+
+import eu.ebrains.kg.common.services.KGServiceClient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.cache.annotation.CacheEvict;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+
+@RequestMapping(value = "/preflight", produces = MediaType.APPLICATION_JSON_VALUE)
```

Comment:
the path is rather generic and doesn't really tell what you're preflighting. We might call the endpoint "authEndpoint" and create an "OPTIONS" endpoint instead of a "GET" endpoint (since we're doing something pretty similar as it's done for CORS). Alternatively, we could also use a "HEAD" endpoint for this.

Reply:
at the same time, we might want to think about renaming the whole class